foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiResponseHandler.java

```diff
@@ -85,19 +94,28 @@ private void buildExceptionAndThrow(@Nonnull final ClassicHttpResponse response)
       throw exception;
     }
 
-    final var maybeError = Try.of(() -> jackson.readValue(content, OpenAiError.class));
+    parseErrorAndThrow(content, exception);
+  }
+
+  /**
+   * Parse the error response and throw an exception.
+   *
+   * @param errorResponse the error response, most likely a JSON of {@link OpenAiError}.
+   * @param baseException a base exception to add the error message to.
+   */
+  static void parseErrorAndThrow(String errorResponse, OpenAiClientException baseException)
+      throws OpenAiClientException {
+    final var maybeError = Try.of(() -> JACKSON.readValue(errorResponse, OpenAiError.class));
     if (maybeError.isFailure()) {
-      exception.addSuppressed(maybeError.getCause());
-      throw exception;
+      baseException.addSuppressed(maybeError.getCause());
+      throw baseException;
     }
 
     final var error = maybeError.get().getError();
     if (error == null) {
-      throw exception;
+      throw baseException;
     }
-    final var message =
-        "Request to OpenAI model failed with %s %s and error message: '%s'"
-            .formatted(response.getCode(), response.getReasonPhrase(), error.getMessage());
-    throw new OpenAiClientException(message);
+    throw new OpenAiClientException(
+        baseException.getMessage() + "and error message: '%s'".formatted(error.getMessage()));
```

Comment:
(Comment/Question)

This exception message recycling doesn't look like good practice to me, have we done that before? Or is this explained somewhere else as good practice?

Reply:
Both answers: no and no.
But this is how it worked before: creating 2 exceptions and reusing the message of the first to create the second.